redun/executors/gcp_utils.py

```diff
@@ -0,0 +1,226 @@
+from enum import Enum
+from typing import Dict, Iterable, List, Tuple, Union
+
+from google.cloud import batch_v1
+from urllib.parse import urlparse
+
+# List of supported available CPU Platforms
+# https://cloud.google.com/compute/docs/instances/specify-min-cpu-platform#availablezones
+class MinCPUPlatform(Enum):
+    XEON_ICE_LAKE = "Intel Ice Lake"
+    XEON_CASCADE_LAKE = "Intel Cascade Lake"
+    XEON_SKYLAKE = "Intel Skylake"
+    XEON_BROADWELL = "Intel Broadwell"
+    XEON_HASWELL = "Intel Haswell"
+    XEON_SANDY_BRIDGE = "Intel Sandy Bridge"
+    EPYC_ROME = "AMD Rome"
+    EPYC_MILAN = "AMD Milan"
+
+
+def get_gcp_client(
+    sync: bool = True,
+) -> Union[batch_v1.BatchServiceClient, batch_v1.BatchServiceAsyncClient]:
+    return batch_v1.BatchServiceClient() if sync else batch_v1.BatchServiceAsyncClient()
+
+
+def batch_submit(
+    client: Union[batch_v1.BatchServiceClient, batch_v1.BatchServiceAsyncClient],
+    job_name: str,
+    project: str,
+    region: str,
+    mount_buckets: List[str],
+    gcs_scratch_prefix: str,
+    machine_type: str,
+    vcpus: int,
```

Comment:
@MSamman I wonder if we can configure jobs to use max available resources by default?

Reply:
Not sure I've seen a way to do so in the batch API.